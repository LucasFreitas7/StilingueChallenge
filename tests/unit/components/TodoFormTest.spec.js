import { shallowMount } from '@vue/test-utils'
import TodoForm from '../../../src/components/TodoForm.vue'


const addMock = jest.fn()
describe('TodoBar.vue', () => {
  const wrapper = shallowMount(TodoForm)
  it('Quando digitado algo no input e apertado enter deve chamar o metodo add', () => {
    wrapper.setMethods({add: addMock})
    const input = wrapper.find('input').trigger('keyup.enter')
    expect(addMock).toHaveBeenCalled()
  })
  it('teste de snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
