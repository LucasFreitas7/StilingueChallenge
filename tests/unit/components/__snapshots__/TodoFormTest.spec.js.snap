// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoBar.vue teste de snapshot 1`] = `
<div class="form-container">
  <div class="todo-check"></div> <input type="text" class="shadow">
</div>
`;
